lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -7,12 +7,14 @@ const experimentalImportMetaResolve = getOptionValue('--experimental-import-meta
  * Generate a function to be used as import.meta.resolve for a particular module.
  * @param {string} defaultParentURL The default base to use for resolution
  * @param {typeof import('./loader.js').ModuleLoader} loader Reference to the current module loader
+ * @param {bool} allowParentURL Whether to permit parentURL second argument for contextual resolution
  * @returns {(specifier: string, parentURL?: string) => string} Function to assign to import.meta.resolve
  */
-function createImportMetaResolve(defaultParentURL, loader) {
-  return function resolve(specifier, parentURL = defaultParentURL) {
+function createImportMetaResolve(defaultParentURL, loader, allowParentURL) {
+  return function resolve(specifier) {
```

Comment:
I think a jsdoc here would be good to call out the shadow argument

Reply:
Can you clarify what you mean here?

Reply:
```suggestion
  /**
   * @param {string} specifier
   * @param {URL['href']} [parentURL] When `--experimental-import-meta-resolve` is specified, a
   * second argument can be provided.
   */
  return function resolve(specifier) {
```

Reply:
Unfortunately the linter does not like this at all.

Reply:
I would like to explicitly retain `import.meta.resolve.length === 1` though.

Reply:
You can set a default value (e.g. `= undefined`) and the length will stay at `1` (hopefully that makes sense)

Reply:
Ah, nice.

Reply:
> You can set a default value (e.g. `= undefined`) and the length will stay at `1` (hopefully that makes sense)

That was my next suggestion 😁